internal/chain/reward.go

```diff
@@ -506,3 +298,343 @@ func waitForCommitSigs(sigsReady chan bool) error {
 	}
 	return nil
 }
+
+func distributeRewardAfterAggregateEpoch(bc engine.ChainReader, state *state.DB, header *block.Header, beaconChain engine.ChainReader,
+	defaultReward numeric.Dec) (reward.Reader, error) {
+	newRewards, payouts :=
+		big.NewInt(0), []reward.Payout{}
+
+	allPayables := []slotPayable{}
+	curBlockNum := header.Number().Uint64()
+
+	allCrossLinks := types.CrossLinks{}
+	startTime := time.Now()
+	for i := curBlockNum - RewardFrequency + 1; i <= curBlockNum; i++ {
+		if i < 0 {
```

Comment:
Because there is this guarantee: ` if bc.Config().IsAggregatedRewardEpoch(header.Epoch())` It should be impossible for i here to be less than 0？

Reply:
for local network where the epoch happens at very early block number such as 20. It is possible.